### PR TITLE
Add skipped failing test for IDE0044

### PR DESCRIPTION
**Pull request by Youssef1313**
The linked issue is now closed and has `Resolution-Fixed` label. So, this false positive should have been fixed.

### CODE REVIEW

**Comment by mavasani** [comment]
@Youssef1313 Seems like IDE0044 is still firing here.

**Comment by Youssef1313** [comment]
Then the issue should be re-opened?
Pinging @sharwell as he worked on that issue.

**Comment by mavasani** [comment]
@Youssef1313 The fix was made in the CodeStyle analyzer NuGet and we need to move to a newer CodeStyle package to get the fix.

**Comment by Youssef1313** [comment]
Aha, got it. Are there any reasons not to update?

**Comment by mavasani** [comment]
Nope, go for it: https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5&package=Microsoft.CodeAnalysis.CSharp.CodeStyle&protocolType=NuGet&version=3.7.0-3.20271.4&view=versions

**Comment by mavasani** [comment]
@Youssef1313 https://github.com/dotnet/roslyn/pull/45288/commits/eeba605d699fa82ec066971e86f817a16a2faf84 seems to be downgrading the version. Can you please upgrade to latest 3.8.x at https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.CodeAnalysis.CSharp.CodeStyle version?

**Comment by Youssef1313** [comment]
@mavasani The failure is still there with the 3.8.x version. What could be the issue?

**Comment by sharwell** [comment]
Updating the code style package is a bit complicated. The dotnet-format version needs to be updated at the same time to a version built from the same source, but the version numbers themselves will be different. I can check with @JoeRobich next week to see what the most recent matching pair is.

**Comment by Youssef1313** [comment]
Using the latest dotnet-format version doesn't seem to work. I'll wait until you check the latest matching pair.

**Comment by Youssef1313** [comment]
Ping @sharwell

**Comment by sharwell** [comment]
@JoeRobich can you verify the matching versions here?

**Comment by JoeRobich** [comment]
> @JoeRobich can you verify the matching versions here?

Currently dotnet-format is tracking the VS 16.7 feed, so to generate a matching version I'll need to update to the 16.8 feed and trigger an update. It will pull in the most recent 3.8.0 build. I'll reply back when a build is ready with dotnet-format version and the M.CA version used.

**Comment by JoeRobich** [comment]
@sharwell @Youssef1313 After updating dotnet-format to consume 3.8.0 builds, here is the matched pair.

Microsoft.CodeAnalysis - 3.8.0-1.20367.11
dotnet-format - 5.0.137002

**Comment by Youssef1313** [comment]
@sharwell @JoeRobich The build is still failing.

**Comment by Youssef1313** [comment]
ping @sharwell @JoeRobich

File: dotnet-tools.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "isRoot": true,
   "tools": {
     "dotnet-format": {
-      "version": "4.0.130203",
+      "version": "5.0.137002",
```
Comment (JoeRobich) [inline dotnet-tools.json:5]:
```suggestion
      "version": "5.0.140801",
```

File: eng/Versions.props [hunk 2]
```diff
@@ -31,7 +31,7 @@
     <MicrosoftCodeAnalysisNetAnalyzersVersion>3.3.0-beta1.20352.4</MicrosoftCodeAnalysisNetAnalyzersVersion>
     <CodeStyleLayerCodeAnalysisVersion>3.6.0-2.final</CodeStyleLayerCodeAnalysisVersion>
     <MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.20378.2</MicrosoftCodeAnalysisTestingVersion>
-    <CodeStyleAnalyzerVersion>3.7.0-3.20271.4</CodeStyleAnalyzerVersion>
+    <CodeStyleAnalyzerVersion>3.8.0-1.20367.11</CodeStyleAnalyzerVersion>
```
Comment (JoeRobich) [inline eng/Versions.props:34]:
```suggestion
    <CodeStyleAnalyzerVersion>3.8.0-2.20407.3</CodeStyleAnalyzerVersion>
```

**Comment by Youssef1313** [comment]
There is now another build error: `error IDE0055: fix formatting` in SymbolDisplayVisitor.Types.cs.
I'm going to fix that, then this should be ready to merge.

**Comment by Youssef1313** [comment]
I just noticed that IDE0044 is still happening, in addition to new IDE0055 appeared after updating the packages.

File: dotnet-tools.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "isRoot": true,
   "tools": {
     "dotnet-format": {
-      "version": "4.0.130203",
+      "version": "5.0.140801",
```
Comment (mavasani) [inline dotnet-tools.json:5]: I am confused on why we are trying to update dotnet-format version in this PR, whose aim is to fix IDE0044 violation coming from CodeStyle analyzer NuGet package. They seem unrelated.

Comment (JoeRobich) [reply dotnet-tools.json:5]: The concern is that bringing in a new CodeStyle package brings in a new FormattingAnalyzer which may use a version of the Roslyn Formatter that differs from what is used in dotnet-format. Meaning running dotnet-format may not fix formatting issues that are reported in the IDE.

**Comment by Youssef1313** [comment]
@mavasani, Any idea why IDE0044 still happens after upgrade?

**Comment by mavasani** [comment]
> Any idea why IDE0044 still happens after upgrade?

Probably would need to debug the analyzer for the test...

**Comment by Youssef1313** [comment]
It would be hard to take the exact class and write a unit test for it to debug that. I'll try to write a smaller unit test hoping the test will fail. If that test passed, I have no idea on what should be the next step.

**Comment by Youssef1313** [comment]
@mavasani, The test I added fails. So, it seems like an analyzer bug.

**Comment by mavasani** [comment]
Can we re-open IDE0044 bug, skip the test with that issue and revert the IDE044 suppression removal? We can then merge this PR and track fixing that issue in a follow-up PR.

**Comment by Youssef1313** [comment]
@mavasani, Sure. But I'll need @JoeRobich help with conflicts regarding the versions.

```
<<<<<<< patch-1
    <MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.20378.2</MicrosoftCodeAnalysisTestingVersion>
    <CodeStyleAnalyzerVersion>3.8.0-2.20407.3</CodeStyleAnalyzerVersion>
=======
    <MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.20407.3</MicrosoftCodeAnalysisTestingVersion>
    <CodeStyleAnalyzerVersion>3.7.0-3.20271.4</CodeStyleAnalyzerVersion>
>>>>>>> master
```

The current master uses a higher version of "MicrosoftCodeAnalysisTestingVersion", but a lower version of "CodeStyleAnalyzerVersion". Should I use master's versions? or my branch versions? or something else?

**Comment by JoeRobich** [comment]
> But I'll need @JoeRobich help with conflicts regarding the versions.

@Youssef1313 If your changes don't rely on updated dependencies then I would use the versions from master.

**Comment by Youssef1313** [comment]
I reverted all version updates in this PR, and skipped the new failing test.
@mavasani, the issue can now be re-opened to address the skipped test.

File: src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs [hunk 4]
```diff
@@ -1705,6 +1705,23 @@ public async Task TestVolatileField3()
 {
     private volatile object first;
     private readonly object second;
+}");
+        }
+
+        [WorkItem(45288, "https://github.com/dotnet/roslyn/pull/45288")]
```
Comment (mavasani) [inline src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs:1711]:
```suggestion
        [WorkItem(45288, "https://github.com/dotnet/roslyn/issues/33009")]
```

File: src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs [hunk 5]
```diff
@@ -1705,6 +1705,23 @@ public async Task TestVolatileField3()
 {
     private volatile object first;
     private readonly object second;
+}");
+        }
+
+        [WorkItem(33009, "https://github.com/dotnet/roslyn/issues/33009")]
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/33009"), Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
```
Comment (sharwell) [inline src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs]: 📝 #33009 relates to ref _returns_, but this one is a new issue related to ref _locals_. Can you file a new issue and update these two links?

Comment (sharwell) [reply src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs]: ➡️ #46785

Comment (Youssef1313) [reply src/Analyzers/CSharp/Tests/MakeFieldReadonly/MakeFieldReadonlyTests.cs]: @sharwell Done.

Do you want me to update the comment for the suppression as well? (That file currently not included in the PR at all)

**Comment by Youssef1313** [comment]
Please squash when merging.

**Comment by 333fred** [comment]
@mavasani @sharwell what is this waiting on?

**Comment by sharwell** [comment]
@333fred it should be ready to manually merge (with a squash commit per https://github.com/dotnet/roslyn/pull/45288#issuecomment-681138842) when the build completes
